File: scratch/fix_logo_bg.py
from PIL import Image
import os

def fix_logo(input_path, output_path, target_color=(248, 245, 240)):
    if not os.path.exists(input_path):
        print(f"File not found: {input_path}")
        return

    print(f"Processing {input_path}...")
    img = Image.open(input_path).convert('RGBA')
    pixels = img.load()

    width, height = img.size
    for y in range(height):
        for x in range(width):
            r, g, b, a = pixels[x, y]
            
            # If the pixel is semi-transparent or part of the "fake transparency" grid
            # Usually fake transparency grids are white/light grey.
            # We want to replace anything that is not "logo" (black/dark) with the target color.
            
            # Calculate intensity
            intensity = (r + g + b) / 3
            
            # If intensity is high (bright), it's background
            if intensity > 150: # Adjust threshold to keep the logo details
                # Replace with target background color, full opacity
                pixels[x, y] = (target_color[0], target_color[1], target_color[2], 255)
            else:
                # Keep original dark pixel (the logo) but make it fully opaque to avoid alpha issues
                # Or keep alpha if it's intended to be semi-transparent black (though logo is usually solid)
                pixels[x, y] = (r, g, b, 255)

    img.save(output_path, 'PNG')
    print(f"Saved fixed logo to {output_path}")

# Run for both original and compressed if they exist
fix_logo('public/logo-v3.png', 'public/logo-v3-fixed.png')
fix_logo('public/logo-v3-compressed.png', 'public/logo-v3-compressed-fixed.png')

File: scratch/fix_logo_bg.js
const { Jimp } = require('jimp');
const path = require('path');
const fs = require('fs');

async function fixLogo(inputPath, outputPath) {
    const fullInputPath = path.resolve(inputPath);
    const fullOutputPath = path.resolve(outputPath);

    if (!fs.existsSync(fullInputPath)) {
        console.log(`File not found: ${fullInputPath}`);
        return;
    }

    console.log(`Processing ${inputPath}...`);
    try {
        const image = await Jimp.read(fullInputPath);
        const { width, height } = image.bitmap;

        // Target background color #F8F5F0
        const targetR = 248;
        const targetG = 245;
        const targetB = 240;

        image.scan(0, 0, width, height, function(x, y, idx) {
            const r = this.bitmap.data[idx + 0];
            const g = this.bitmap.data[idx + 1];
            const b = this.bitmap.data[idx + 2];
            const a = this.bitmap.data[idx + 3];

            const intensity = (r + g + b) / 3;

            // If it's a light pixel (part of the grid or white background)
            if (intensity > 180 || a < 100) {
                this.bitmap.data[idx + 0] = targetR;
                this.bitmap.data[idx + 1] = targetG;
                this.bitmap.data[idx + 2] = targetB;
                this.bitmap.data[idx + 3] = 255; // Make it opaque
            } else {
                // Keep it as is (likely the dark logo part)
                // But ensure it's opaque to avoid grid showing through semi-transparency
                this.bitmap.data[idx + 3] = 255;
            }
        });

        await image.write(fullOutputPath);
        console.log(`Saved fixed logo to ${outputPath}`);
    } catch (err) {
        console.error(`Error processing ${inputPath}:`, err);
    }
}

async function run() {
    await fixLogo('public/logo-v3.png', 'public/logo-v3-fixed.png');
    await fixLogo('public/logo-v3-compressed.png', 'public/logo-v3-compressed-fixed.png');
}

run();
